crowbar_framework/lib/crowbar/checks/network.rb

```diff
@@ -0,0 +1,142 @@
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+require "resolv"
+require "ipaddr"
+require "socket"
+require "logger"
+
+module Crowbar
+  module Checks
+    class Network
+      class << self
+        def ip_addrs(fqdn, version = :ipv4)
+          ip_addrs = []
+
+          Resolv.getaddresses(fqdn).each do |address|
+            ip_addr = IPAddr.new(address)
+            if version == :ipv6
+              ip_addrs.push address if ip_addr.ipv6?
+            else
+              ip_addrs.push address if ip_addr.ipv4?
+            end
+          end
+
+          ip_addrs
+        end
+      end
+
+      def fqdn_detected?
+        if hostname.blank? || domain.blank? || fqdn.blank?
+          return false
+        end
+
+        true
+      end
+
+      def ip_resolved?
+        if ipv4_addrs.empty? && ipv6_addrs.empty?
+          return false
+        end
+
+        true
+      end
+
+      def loopback_resolved?
+        if ipv4_addrs.find { |e| /^127/ =~ e } || ipv6_addrs.find { |e| /^::[0-9]$/ =~ e }
+          return false
+        end
+
+        true
+      end
+
+      def ip_configured?
+        ipv4_configured = false
+        ipv6_configured = false
+
+        addr_infos = Socket.ip_address_list
+        addr_infos.each do |addr_info|
+          ipv4_configured = true if ipv4_addrs.include?(addr_info.ip_address)
+          ipv6_configured = true if ipv6_addrs.include?(addr_info.ip_address)
+        end
+
+        unless ipv4_configured || ipv4_addrs.empty?
+          return false
+        end
+        unless ipv6_configured || ipv6_addrs.empty?
+          return false
+        end
+
+        true
+      end
+
+      def network_valid?
+        if File.exist?("/etc/crowbar/network.json")
+          network_json = "/etc/crowbar/network.json"
+        else
+          network_json = Rails.root.join(
+            "..", "chef", "data_bags", "crowbar", "template-network.json").to_s
+        end
+
+        admin_ip = NodeObject.admin_node.ip
+        unless ipv4_addrs.include?(admin_ip)
+          return false
+        end
+        unless system("network-json-validator --admin-ip \"#{admin_ip}\" #{network_json}")
+          return false
+        end
+
+        true
+      end
+
+      def firewall_disabled?
+        if system("LANG=C iptables -n -L | grep -qvE '^$|^Chain [^ ]|^target     prot'")
+          return false
+        end
+
+        true
+      end
+
+      def ping_succeeds?
+        unless system("ping -c 1 #{fqdn}")
```

Comment:
Don't we want to redirect the output to /dev/null? Or am I guessing wrong that we'll see ping output on every start of the service?